### PR TITLE
Last minute API fixes

### PR DESCRIPTION
**Pull request by lukechampine**
API.md had some errors. siac was implementing a couple calls wrong, and downloadqueue wasn't implemented at all. There was also a `Synchronize` error where synchronizing to a peer with the same height would cause an error.

### CODE REVIEW

File: modules/gateway/synchronize.go [hunk 1]
```diff
@@ -83,14 +82,21 @@ func (g *Gateway) sendBlocks(conn modules.NetConn) (err error) {
...
 	}
-	if !found {
-		// The genesis block should be included in knownBlocks - if no matching
-		// blocks are found, the caller is probably on a different blockchain
-		// altogether.
-		return errors.New("no matching block found")
+	// If we didn't find any matching blocks, or if we're already
+	// synchronized, don't send any blocks. The genesis block should be
+	// included in knownBlocks, so if no matching blocks are found, the caller
+	// is probably on a different blockchain altogether.
```
Comment (DavidVorick) [inline modules/gateway/synchronize.go]: hmm. Shouldn't knownBlocks[0] be a match if the two are already synchronized?

File: siad/API.md [hunk 2]
```diff
@@ -170,28 +173,31 @@ windowSize   int
...
+`minFilesize is the minimum allowed file size.
 
-maxFilesize is the maximum allowed file size.
+`maxFilesize is the maximum allowed file size.
 
-minDuration is the minimum amount of time a contract is allowed to last.
+`minDuration is the minimum amount of time a contract is allowed to last.
 
-maxDuration is the maximum amount of time a contract is allowed to last.
+`maxDuration is the maximum amount of time a contract is allowed to last.
```
Comment (DavidVorick) [inline siad/API.md]: you didn't close the hooks on some of these

File: siad/API.md [hunk 3]
```diff
@@ -388,7 +411,13 @@ Parameters: none
...
+	FullBalance  int
+	NumAddresses int
 }
 ```
-Balance is the spendable balance of the wallet.
+`Balance` is the spendable balance of the wallet.
+
+`FullBalance` is the balance of the wallet, including unconfirmed coins.
+
+`NumAddresses is the number of addresses controlled by the wallet.
```
Comment (DavidVorick) [inline siad/API.md]: didn't close the hook here either

**Comment by DavidVorick** [comment]
done reviewing

File: modules/gateway/synchronize.go (hunk 1 above)
Comment (lukechampine) [inline modules/gateway/synchronize.go]: yes, so `found` will be true, but `start` will be greater than `g.state.Height()` (actually it should be exactly `g.state.Height() + 1`, but no harm in being safe).

**Pull request merged**